inc/zoo/swar/SWAR.h

```diff
@@ -118,6 +122,16 @@ struct SWAR {
         return SWAR(value() >> (NBits * laneCount));
     }
 
+    /// \param protectiveMask should clear the bits that would cross the lane
+    constexpr SWAR shiftIntraLaneLeft(int bitCount, SWAR protectiveMask) const noexcept {
```

Comment:
It seems like the protective mask can be constexpr known by the SWAR object, so why is it part of the method signature?

Reply:
Yes, this is the right signature.
The bits that will be cleared are directly related to the count of shifts, it is natural to maintain the protective mask by the caller, otherwise, the mask will be computed on all invocations.  Not sure the compiler would maintain this mask somewhere to not calculate it.
Do you agree the optimizer might not be good enough?

Reply:
Good explanation, should probably include something like it in the code.